src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -126,12 +128,14 @@ export class ConnectionWidget {
 	}
 
 	public createConnectionWidget(container: HTMLElement): void {
-		this._serverGroupOptions = [this.DefaultServerGroup];
-		this._serverGroupSelectBox = new SelectBox(this._serverGroupOptions.map(g => g.name), this.DefaultServerGroup.name, this._contextViewService, undefined, { ariaLabel: this._serverGroupDisplayString });
-		this._previousGroupOption = this._serverGroupSelectBox.value;
+		if (this._providerName !== Constants.cmsProviderName) {
```

Comment:
What if you at least made a class that extended ConnectionWidget which implemented the correct behavior and then had the service use that if it was specified as a CMS connection. As discussed in the earlier iteration of the review having this kind of hard-coded extension-specific stuff in a core component like the connection dialog is very undesirable.